### DISCUSSION TITLE
No intellisense or go to definition in templates for Vue 2.7?

### DISCUSSION THREAD

**Discussion by douglasg14b** [vuejs/language-tools#1644] [Q&A] [answered]
I'm trying to do a stepped upgrade to Vue3, and using Volar. I've upgraded to Vue 2.7, and followed the Vue 2 instructions for getting set up with Volar. I'm trying to migrate my `class-component` components over to `<script setup>` composition API ones.

However, it seems there is no connection between the script and the template, at least as far as autocomplete, intellisense, or go-to-definition is concerned. There is no go to definition for imported components, or intellisense for components...etc However, there is auto-imports for components...?

It does correctly handle `noUnsusedLocals`, so there is something happening there.

Edit: Also doesn't work with `defineComponent` either :/

**Example:**

![image](https://user-images.githubusercontent.com/1400380/181424745-8164824e-f49c-4f7e-8167-2dd01d733de3.png)

![image](https://user-images.githubusercontent.com/1400380/181630726-37183a5e-f352-4c15-b186-212a157d9346.png)

**Comment by kingyue737** [comment, marked as answer]
Volar works perfect both for my Vue 2.6 and Vue 2.7 projects. Maybe something is wrong in your `tsconfig` or other settings.
You can init a new project by 
```
npm init vue@2
```
And compare your settings with official starter template
